components/usage/pkg/apiv1/billing.go

```diff
@@ -299,7 +299,10 @@ func (s *BillingService) FinalizeInvoice(ctx context.Context, in *v1.FinalizeInv
 		logger.WithError(err).Errorf("Failed to insert Invoice usage record into the db.")
 		return nil, status.Errorf(codes.Internal, "Failed to insert Invoice into usage records.")
 	}
-
+	_, err = s.ccManager.IncrementBillingCycle(ctx, usage.AttributionID)
+	if err != nil {
+		return nil, err
```

Comment:
This potentially leaks a DB error to the API. You should be catching the error and using `status.Errorf(codes.NotFound, "Failed to get invoice with ID %s: %s", in.GetInvoiceId(), err.Error())` report an appropriate API facing error + code